src/main/java/com/aws/greengrass/lifecyclemanager/KernelLifecycle.java

```diff
@@ -103,9 +127,47 @@ public void launch() {
             kernel.getContext().get(c).startup();
         }
 
+        final Queue<DeviceIdentityInterface> provisioningPlugins = findProvisioningPlugins();
```

Comment:
I see that the provisioning plugins are also going to be under the `services` config namespace, are these plugins going to be dependencies of `main` ? If so they won't load in `startupAllServices` and are also going to fail or show up in the result of `orderedDependencies`. Should we filter instances of DeviceIdentityInterface from those relevant places?

Reply:
They are not going to be dependencies of main, unless they are annotated with ImplementsService (and they extend GreengrassService). Plugins are under the services config namespace and as far as I understand, there can be components under that namespace which are not direct/indirect dependencies of main. If this is not true then we may need a different namespace for provisioning plugins. For now I do not see a need for that.

Reply:
Okay, just calling out that this needs to be given some thought. If these plugins are allowed to be updated via deployments then they will become dependencies of main given how deployments work and it will introduce some complications here

Reply:
That brings out a good point. Although we do not plan to support cloud deployment right now but when we start supporting reprovisioning, we will need to support updating plugin config via cloud deployments. We need to make sure anything done here does not complicate reprovisioning in future. I will think through, discuss and update.